pytorch_lightning/plugins/environments/slurm_environment.py

```diff
@@ -100,6 +85,25 @@ def detect() -> bool:
         """Returns ``True`` if the current process was launched on a SLURM cluster."""
         return "SLURM_NTASKS" in os.environ
 
+    @staticmethod
+    def job_name() -> Optional[str]:
+        return os.environ.get("SLURM_JOB_NAME")
+
+    @staticmethod
+    def job_id() -> Optional[int]:
+        job_id = os.environ.get("SLURM_JOB_ID")
+        if job_id:
+            try:
+                job_id = int(job_id)
+            except ValueError:
+                job_id = None
+
+        # in interactive mode, don't make logs use the same job id
+        in_slurm_interactive_mode = SLURMEnvironment.job_name() == "bash"
+        if in_slurm_interactive_mode:
+            job_id = None
+        return job_id
```

Comment:
@awaelchli why do we do this logic inside of the job id call instead of at the log time?

Reply:
yes that's what I meant in the comment https://github.com/PyTorchLightning/pytorch-lightning/pull/10622#discussion_r752622231 over here, as we can disentangle and simplify this. 
The code was already like this before we moved it to SLURMEnvironment.

Reply:
Should we do it in this PR?

Reply:
No, because this PR is focusing on adding the job_name, but the comment is appreciated and definitely an opportunity for a new PR.